### ISSUE TITLE
[BUG] Exposing a blackboard entry as prefab property doesn't work

### ISSUE THREAD

**Issue by jankrassnigg** [ezEngine/ezEngine#829]
When a prefab contains a blackboard component, you can expose the "initial value" element as a prefab parameter.

During instantiation this fails, though, the externally set value cannot get applied.

It goes through ezPropertyPath::SetValue and then I guess through ezReflectionUtils::SetArrayPropertyValue where it somehow isn't able to find the desired slot.

More specifically in ezPropertyPath::ResolvePath under case ezPropertyCategory::Array it fails here:

```
const ezUInt32 uiIndex = path[0].m_Index.ConvertTo<ezUInt32>();
if (uiIndex >= pSpecific->GetCount(pCurrentObject))
  return EZ_FAILURE;
```

To debug it from the top, put a breakpoint into ezPrefabResource::ApplyExposedParameterValues.

**Comment by Sanakan8472** [comment]
The reason for this is that the engine prefab instance of the ezBlackboardComponent component has an empty m_InitialEntries array as the engine prefab instance goes through the ezBlackboardComponent::DeserializeComponent and that omits the array.
Thus, the property path becomes invalid as the array index is invalid and the path can't be resolved.

**Comment by jankrassnigg** [comment]
Because the prefab uses the default value? Would it work if the prefab already had a different value assigned to the blackboard entry?

**Comment by Sanakan8472** [comment]
No, the problem is this code in `ezBlackboardComponent::DeserializeComponent`:
```
  // we don't write the data to m_InitialEntries, because that is never needed anymore at runtime
  ezDynamicArray<ezBlackboardEntry> initialEntries;
  if (s.ReadArray(initialEntries).Succeeded())
  {
    for (auto& entry : initialEntries)
    {
      m_pBoard->RegisterEntry(entry.m_sName, entry.m_InitialValue, entry.m_Flags);
    }
  }
```
The member array `m_InitialEntries` is not populated and instead data is sent to `m_pBoard` instead. So any attempt to change `m_InitialEntries` later will fail as it is empty.

**Comment by jankrassnigg** [comment]
Interesting. Need to think about this a bit. We could maybe redirect attempts to write to it at runtime.
